### ISSUE TITLE
user.upload() does not accept Blob

### ISSUE THREAD

**Issue by ThisIsMissEm** [testing-library/user-event#923]
### Reproduction example

n/a

### Prerequisites

Create a Blob with: 
```
const blob = new Blob(["foo"], {
  type: "text/plain",
});
```

Then try passing to:
```
user.upload(getByTestId("upload-input"), [blob])
```

### Expected behavior

The blob is uploaded as if it were a File

### Actual behavior

Typescript errors out on `Blob` not being compatible with `File` or `File[]`, despite `File` being based on `Blob`: https://developer.mozilla.org/en-US/docs/Web/API/Blob

### User-event version

14.1.0

### Environment

```
"@testing-library/react": "^11.2.6",
"@testing-library/react-hooks": "^7.0.2",
"@testing-library/user-event": "^14.1.0",
"jest": "^26.6.3",
"react": "^17.0.2",
"react-dom": "^17.0.2",
```

### Additional context

This appears to just be a types issue

**Comment by ph-fritsche** [comment]
I'm not sure if I understand what this issue is about. `.upload()` accepts `File | File[]` as third parameter.
`Blob` does not extend `File`, `File` extends `Blob`. So a `Blob` object does not satisfy the type.

**Comment by ThisIsMissEm** [comment]
Correct, but is there a reason why a Blob could not be uploaded? Or would I need to create a File for that blob?

**Comment by ph-fritsche** [comment]
This library is about simulating user interaction, in this case selecting `files` for an `<input type="file">`.
Yes, you need to supply `File` objects, just like using the file selector in the browser does.
